googleapiclient/http.py

```diff
@@ -82,22 +89,65 @@ def _retry_request(http, num_retries, req_type, sleep, rand, uri, method, *args,
     resp, content - Response from the http request (may be HTTP 5xx).
   """
   resp = None
+  content = None
   for retry_num in range(num_retries + 1):
     if retry_num > 0:
-      sleep(rand() * 2**retry_num)
+      # Sleep before retrying.
+      sleep_time = rand() * 2 ** retry_num
       logging.warning(
-          'Retry #%d for %s: %s %s%s' % (retry_num, req_type, method, uri,
+          'Sleeping %.2f seconds before retry %d of %d for %s: %s %s%s' %
+          (sleep_time, retry_num, num_retries, req_type, method, uri,
           ', following status: %d' % resp.status if resp else ''))
+      sleep(sleep_time)
 
     try:
       resp, content = http.request(uri, method, *args, **kwargs)
-    except ssl.SSLError:
+    # Retry on SSL errors and socket timeout errors.
+    except (ssl.SSLError, TimeoutError) as ex:
```

Comment:
Code of the form
```
except (<exception type 1>, <exception type 2>) as <name>:
  if isinstance(<name>, <exception type>):
    <statements>
```
invites the question of whether it can be of the form
```
except <exception type 1> as <name>:
  <statements>
except <exception type 2> as <possibly other name>:
  <statements>
```
. Does that question apply here? I'm having trouble seeing.